src/main/java/eu/execom/pomodoroTeam/services/TeamService.java

```diff
@@ -46,4 +55,53 @@ public TeamEntity updateTeam(Long id, String name) {
         }
         return teamRepository.save(team);
     }
-}
+
+    public Invitation getByActivationLink(String activationLink) {
+        return invitationRepository.findOneByActivationLink(activationLink);
+    }
+
+    public NewInvitationDto DataFromInvitation(String activationLink) {
+        Invitation invitation = getByActivationLink(activationLink);
+
+        NewInvitationDto newInvitationDto = new NewInvitationDto();
+        newInvitationDto.setEmail(invitation.getUsername());
+        newInvitationDto.setTeamId(invitation.getTeamId());
+        invitationRepository.delete(invitation);
+
+        return newInvitationDto;
+    }
+
+    public TeamEntity getSingleTeam(Long id) {
+        return teamRepository.getOne(id);
+    }
+
+    public boolean checkIfUserExistsInThatTeamByEmail(String email, Long teamId) {
+        TeamEntity team = teamRepository.getOne(teamId);
+        List<UserEntity> users = team.getUsers();
+        boolean found = false;
+        for (UserEntity u : users) {
+            if (u.getEmail().equals(email)) {
+                found = true;
+            }
+        }
+        return found;
+    }
+
+    public TeamWithUserDto checkIfUserExistsInAnyTeam(String email, Long userId) {
```

Comment:
You can use just email or userId, no need to use both.
Why do you need this check at all? 
If there are many teams, with many users, this will be very expensive method. Could you create a query that returns a list of teams in which user belongs, and just return true or false if it has records or the result is empty?